lib/srv/db/cloud/watchers/rds.go

```diff
@@ -268,24 +287,96 @@ func (f *rdsAuroraClustersFetcher) String() string {
+	engines, err := filterByRecognizedRDSEngines(ctx, rdsClient, supportedEngines, maxPages)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 	return []*rds.Filter{{
-		Name: aws.String("engine"),
-		Values: aws.StringSlice([]string{
-			services.RDSEnginePostgres,
-			services.RDSEngineMySQL,
-			services.RDSEngineMariaDB}),
-	}}
+		Name:   aws.String("engine"),
+		Values: aws.StringSlice(engines),
+	}}, nil
 }
 
 // auroraFilters returns filters to make sure DescribeDBClusters call returns
 // only databases with engines Teleport supports.
-func auroraFilters() []*rds.Filter {
+func auroraFilters(ctx context.Context, rdsClient rdsiface.RDSAPI, checkFilters bool, maxPages int) ([]*rds.Filter, error) {
+	supportedEngines := []string{
+		services.RDSEngineAurora,
+		services.RDSEngineAuroraMySQL,
+		services.RDSEngineAuroraPostgres,
+	}
+	if !checkFilters {
+		return []*rds.Filter{{
+			Name:   aws.String("engine"),
+			Values: aws.StringSlice(supportedEngines),
+		}}, nil
+	}
+
+	// filter supported engines to include only the ones recognized by AWS in this region.
+	engines, err := filterByRecognizedRDSEngines(ctx, rdsClient, supportedEngines, maxPages)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
 	return []*rds.Filter{{
-		Name: aws.String("engine"),
-		Values: aws.StringSlice([]string{
-			services.RDSEngineAurora,
-			services.RDSEngineAuroraMySQL,
-			services.RDSEngineAuroraPostgres}),
-	}}
+		Name:   aws.String("engine"),
+		Values: aws.StringSlice(engines),
+	}}, nil
+}
+
+// filterByRecognizedRDSEngines filters Teleport supported engine names to include only those recognized by AWS.
+func filterByRecognizedRDSEngines(ctx context.Context, rdsClient rdsiface.RDSAPI, supportedEngines []string, maxPages int) ([]string, error) {
+	recognized, err := getRecognizedRDSEngines(ctx, rdsClient, maxPages)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// now filter for supported + recognized engine names.
+	var engines []string
+	for _, engine := range supportedEngines {
+		if _, ok := recognized[engine]; ok {
+			engines = append(engines, engine)
+		}
+	}
+	// important that we don't return 0 engine names, as that would apply no filtering at all in AWS DescribeDBClusters/Instances calls.
+	if len(engines) == 0 {
+		return nil, trace.NotFound("Teleport supports engine names %v but none are recognized in this region.", supportedEngines)
+	}
+	return engines, nil
+}
+
+// getRecognizedRDSEngines gets all engine versions within an AWS region and builds a string set of AWS recognized engine names.
+func getRecognizedRDSEngines(ctx context.Context, rdsClient rdsiface.RDSAPI, maxPages int) (map[string]struct{}, error) {
+	var engines []string
+	var pageNum int
+	// https://docs.aws.amazon.com/sdk-for-go/api/service/rds/#RDS.DescribeDBEngineVersionsPages
+	err := rdsClient.DescribeDBEngineVersionsPagesWithContext(ctx, &rds.DescribeDBEngineVersionsInput{},
```

Comment:
hard to say if this takes longer than the "try each engine one at a time" approach because it probably depends on how many instances/clusters you have in AWS. I believe each page is 100 items

Reply:
well if there are a lot instances/clusters, the next describe-call after the get-engine-version-call will be slow too 😉